### PR TITLE
Fix panic in PointerDB.Get

### PR DESCRIPTION
**Pull request by kaloyan-raev**

### CODE REVIEW

File: pkg/pointerdb/pointerdb.go [hunk 1]
```diff
@@ -153,6 +153,7 @@ func (s *Server) Get(ctx context.Context, req *pb.GetRequest) (resp *pb.GetRespo
 		node, err := s.cache.Get(ctx, piece.NodeId)
 		if err != nil {
 			s.logger.Error("Error getting node from cache")
+			continue
```
Comment (egonelbre) [inline pkg/pointerdb/pointerdb.go]: Didn't receiver assume that the piece num and node index match in the ndoes?

Comment (kaloyan-raev) [reply pkg/pointerdb/pointerdb.go]: You are right. Pushed another commit.

Comment (jtolds) [reply pkg/pointerdb/pointerdb.go]: yeah

File: pkg/storage/segments/store.go [hunk 2]
```diff
@@ -363,7 +363,9 @@ func lookupAndAlignNodes(ctx context.Context, oc overlay.Client, nodes []*pb.Nod
 		}
 	}
 	for _, v := range nodes {
-		v.Type.DPanicOnInvalid("lookup and align nodes")
+		if v != nil {
```
[the remark this answers is not in the document]
Comment (kaloyan-raev) [reply pkg/storage/segments/store.go]: I don't find it.

[the remark this answers is not in the document]
Comment (mniewrzal) [reply pkg/storage/segments/store.go]: it's in pkg/storage/segments/repairer.go

[the remark this answers is not in the document]
Comment (kaloyan-raev) [reply pkg/storage/segments/store.go]: I see there only invocation to `lookupAndAlignNodes`, but not a definition of such function. There is only one `lookupAndAlignNodes` in our code base.

[the remark this answers is not in the document]
Comment (jtolds) [reply pkg/storage/segments/store.go]: ohhhhh i thought for sure the repairer was in a different package. okay

**Pull request merged**